### ISSUE TITLE
Relative Markdown Link Completion

### ISSUE THREAD

**Issue by fingolfin19** [artempyanykh/marksman#182]
Follow up to https://github.com/artempyanykh/marksman/issues/27#issuecomment-1369911689 since I do not see that a new issue was ever created.

> When i start typing [title]( then absolute paths are suggested like /path-to-file.md this works fine.
Also autocompletion works, when using # afterwards - like [title](/path-to-file.md# - to get a link to a header.

> For relative paths both of this does not work.
When typing [title](./ nothing further is being suggested by marksman.
After finishing the path to an existing file and starting to type a header - like [title](./path-to-file.md# - no completions are suggested.

**Comment by artempyanykh** [comment]
Thanks for filing this issue @fingolfin19! I definitely got lost in #27

**Comment by artempyanykh** [comment]
To leave some relevant context here:
1. Resolving links works both with absolute and with relative paths.
2. We can make completion follow the user's style, e.g. if the user starts entering a relative path we can suggest completions that would also be relative paths.

**Comment by gustavopcassol** [comment]
Hey @artempyanykh. I've been trying to stick to relative paths for compatibility with jetbrains tools' markdown preview. Is there support for relative completion today?

**Comment by serranomorante** [comment]
Hi! I think it is not supported yet

![image](https://github.com/artempyanykh/marksman/assets/43413954/d964658a-dde2-467e-80a0-8f7b481230e7)

**Comment by serranomorante** [comment]
> ~Hi! I think it is not supported yet~
> 
> ![image](https://user-images.githubusercontent.com/43413954/260326837-d964658a-dde2-467e-80a0-8f7b481230e7.png)

I correct, `Link to non-existent document` warning doesn't appear if I also open those 2 files in a buffer. I don't know if this is the intended behaviour for links to work only on opened buffers.

**Comment by artempyanykh** [comment]
@serranomorante not really. Although, it depends on how your project folder is organized. Is your root folder under version control? If not could you `git init` inside the root folder and try this again?

**Comment by artempyanykh** [comment]
Hi @gustavopcassol ! Unfortunately, too much *life* in the way of open source -- I didn't have time to implement this. It's still on the todo list, though. Also, PRs are welcome :)

**Comment by serranomorante** [comment]
> @serranomorante not really. Although, it depends on how your project folder is organized. Is your root folder under version control? If not could you `git init` inside the root folder and try this again?

I will check on this, thank you for your response!

**Comment by serranomorante** [comment]
I'm so sorry for having branch-off the main point here. My project folder indeed was in version control but I had a very "special" `.gitignore` file and I just read that marksman [behaves like git](https://github.com/artempyanykh/marksman/blob/main/docs/features.md#ignore-files) in that sense. The solution was adding a `.ignore` file as recommended [here](https://github.com/artempyanykh/marksman/issues/113) and that solved all of those warnings about relative links.

Thank you!

**Comment by artempyanykh** [comment]
@serranomorante happy to hear the problem is resolved!

**Comment by jhrv** [comment]
Using [mkdocs](https://www.mkdocs.org/) it's recommended using relative links, so having completion with a list of suggestions relative to the current file when you press "dot" after the start parenthesis would be fantastic.

**Comment by carried-away** [comment]
If I was good enough, I would certainly do a PR. but I'll put a $50 bounty on this is anyone is interested.

**Comment by eread** [comment]
> Using [mkdocs](https://www.mkdocs.org/) it's recommended using relative links, so having completion with a list of suggestions relative to the current file when you press "dot" after the start parenthesis would be fantastic.

@jhrv @artempyanykh Being able to having Marksman work on link completion always relative to the current file (without leading dots or slashes) would also be fantastic. There must be a lot of Markdown-based projects like that.

It's not possible to configure that way today, is it?
